### ISSUE TITLE
FilterVcf blows up with an IAE

### ISSUE THREAD

**Issue by akiezun** [broadinstitute/gatk#1728]
```
./gatk-launch FilterVcf -I src/test/resources/large/dbsnp_138.b37.20.21.vcf -O a.vcf
```

```
java.lang.IllegalArgumentException: A reference dictionary is required for creating Tribble indices on the fly
    at htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder.build(VariantContextWriterBuilder.java:411)
    at org.broadinstitute.hellbender.tools.picard.vcf.filter.FilterVcf.doWork(FilterVcf.java:71)
    at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:102)
    at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:155)
    at org.broadinstitute.hellbender.cmdline.PicardCommandLineProgram.instanceMain(PicardCommandLineProgram.java:61)
    at org.broadinstitute.hellbender.Main.instanceMain(Main.java:67)
    at org.broadinstitute.hellbender.Main.main(Main.java:82)
```

This file does have a sequence dictionary and the Picard original FilterVcf does not blows up on this file but does blow up if file has no sequence dictionary, eg
`java -jar ~/bin/picard-tools-2.0.1/picard.jar FilterVcf I=src/test/resources/empty.vcf O=a.vcf`

```
Exception in thread "main" picard.PicardException: The input vcf must have a sequence dictionary in order to create indexed vcf or bcfs.
    at picard.vcf.filter.FilterVcf.doWork(FilterVcf.java:106)
    at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:209)
    at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:95)
    at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:105)
```

**Comment by akiezun** [comment]
blocked on #1210

**Comment by droazen** [comment]
@akiezun If you think the problem might extend beyond `FilterVcf`, and that our sequence dictionary detection might be broken in general in some way for vcfs, could you create a separate ticket to investigate, and assign to @cmnbroad?

**Comment by akiezun** [comment]
depends on if it's limited to picard ports or no

**Comment by droazen** [comment]
Closing as obsolete
